Frontend/Gmfctn/src/app/core/services/thank-service/thank.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, EMPTY, Observable } from 'rxjs';
+import { catchError, map, tap } from 'rxjs/operators';
+import { Thank } from 'src/app/shared/models/thank';
+import { apiUrl } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ThankService implements OnDestroy {
+  thank$: BehaviorSubject<Thank> = new BehaviorSubject(null as unknown as Thank);
+
+  constructor(private httpClient: HttpClient) { }
+
+  ngOnDestroy(): void {
+    this.thank$.complete();
+  }
+
+  sendThank(id: string, message: string): Observable<any> {
+    return this.httpClient.post<any>(`${apiUrl}api/thank?Text=${message}&ToUserId=${id}`, '')
+    .pipe();
+  }
+
+  getLastThank(): Observable<Thank> {
+    return this.httpClient.get<Thank>(`${apiUrl}api/thank`)
+    .pipe(map(thank => {
+      const thankWithUserAvatar = thank;
+
+      if ( thankWithUserAvatar ) {
+      this.httpClient
```

Comment:
I'd suggest to have a separate files service and call it wherever you'll need to get the file, as currently this logic is duplication appearing from service to service

Reply:
Fixed 😎